src/garage/envs/gym_env.py

```diff
@@ -0,0 +1,409 @@
+"""Wrapper class that converts gym.Env into GymEnv."""
+
+import copy
+import math
+
+import akro
+import gym
+from gym.wrappers.time_limit import TimeLimit
+
+from garage import Environment, EnvSpec, EnvStep, StepType
+
+# The gym environments using one of the packages in the following lists as
+# entry points don't close their viewer windows.
+KNOWN_GYM_NOT_CLOSE_VIEWER = [
+    # Please keep alphabetized
+    'gym.envs.atari',
+    'gym.envs.box2d',
+    'gym.envs.classic_control'
+]
+
+KNOWN_GYM_NOT_CLOSE_MJ_VIEWER = [
+    # Please keep alphabetized
+    'gym.envs.mujoco',
+    'gym.envs.robotics'
+]
+
+
+def _get_env_time_limit(env):
+    """Get time limit from a gym.Env.
+
+    Args:
+        env (gym.Env): the input gym.Env
+
+    Returns:
+        int: if there max_episode_length is found in the gym.Env. Or None if
+        not found.
+
+    Raises:
+        RuntimeError: if the gym.Env is wrapped by a gym.TimeLimit,
+        and env.spec._max_episode_steps and env._max_episode_steps don't match.
+    """
+    gym_spec_steps = None
+    if hasattr(env, 'spec') and env.spec:
+        gym_spec_steps = env.spec.max_episode_steps
+    else:
+        # metaworld env doesn't have spec, what to do?
+        pass
+
+    if isinstance(env, TimeLimit):
+        # pylint: disable=protected-access
+        timelimit_steps = env._max_episode_steps
+        if gym_spec_steps and gym_spec_steps != timelimit_steps:
+            raise RuntimeError('Expect wrapped gym environment '
+                               ' TimeLimit._max_episode_steps '
+                               'be equal to env.spec.max_episode_stpes'
+                               '({}), but got {} instead'.format(
+                                   timelimit_steps, gym_spec_steps))
+
+    return gym_spec_steps
+
+
+class GymEnv(Environment):
+    """Returns an abstract Garage wrapper class for gym.Env.
```

Comment:
Please make sure this renders nicely on ReadTheDocs. it is likely to be frequently-visited docstring.